templates/cert-watcher

```diff
@@ -2,6 +2,20 @@
 
 # This script watches for new SSL certificates and installs them to haproxy.
 
+install_cert() {
+    # Certbot behaves differently when renewing a certificate as opposed to
+    # obtaining one for the first time.  The only way I've found to make sure
+    # all files are fully written is to wait some time before actually
+    # installing the certificate.
+    sleep 10
+    cd "{{ LETSENCRYPT_LIVE_DIR }}/$1"
+    # Haproxy expects the certificate and the private key concatenated
+    # together to a single .pem file.  We lock the certs directory to protect
+    # against haproxy reading a half-written certificate.
+    flock {{ LOAD_BALANCER_CERTS_DIR }} cat fullchain.pem privkey.pem \
```

Comment:
Yes, that part of the operation is serial, but it basically takes no time anyway.  The slow operation that should happen in parallel is the `sleep`.

I know it seems slightly stupid to have multiple process wait in parallel.  However, I don't know a better solution:  After observing the link creation, we have to wait a bit to be sure that everything is flushed to disk.  These wait periods definitely don't need to be serialised, as they currently are.